src/com/tictacgo/TicTacGoGameActivity.java

```diff
@@ -0,0 +1,279 @@
+package com.tictacgo;
+
+import android.app.Activity;
+import android.app.FragmentManager;
+import android.app.FragmentTransaction;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.tictacgo.DirectionPickerFragment.OnDirectionPickedListener;
+import com.tictacgo.data.Board;
+import com.tictacgo.data.Board.Player;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TicTacGoGameActivity extends Activity implements OnDirectionPickedListener {
+  /**
+   * The Board of the game
+   */
+  private Board board;
+
+  /**
+   * The FrameLayout in which the game is to be drawn
+   */
+  private FrameLayout fl;
+
+  /**
+   * OnClickListener to use for each piece
+   */
+  private View.OnClickListener pieceClicked;
+
+  /**
+   * The height of the usable screen area
+   */
+  private int height;
+
+  /**
+   * An ArrayList of the Boards for undo and redo
+   */
+  private List<Board> undoHistory;
+
+  /**
+   * The index of the current undo/redo history
+   */
+  private int historyIndex;
+
+  /**
+   * The initial turn selection. Used for the New Game Button
+   */
+  private Player turn;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_tic_tac_go_game);
+
+    Intent intent = getIntent();
+
+    turn = (Player) intent.getSerializableExtra(TicTacGoMenuActivity.playerKey);
+
+    // Undo/redo initialization
+    undoHistory = new ArrayList<>();
+    historyIndex = 0;
+
+    height = intent.getIntExtra(TicTacGoMenuActivity.heightKey, 300);
+    board = new Board(turn, height, getBaseContext());
+
+    // Set up the screen
+    ((TextView) findViewById(R.id.gamePlayerOneName))
+        .setText(intent.getStringExtra(TicTacGoMenuActivity.p1NameKey));
+    ((TextView) findViewById(R.id.gamePlayerTwoName))
+        .setText(intent.getStringExtra(TicTacGoMenuActivity.p2NameKey));
+    fl = (FrameLayout) findViewById(R.id.gameBoard);
+
+    // What to do when an empty space is clicked
+    pieceClicked = new View.OnClickListener() {
+      public void onClick(View v) {
+        // Create new Fragment Transaction and remove all previous DirectionPickers
+        FragmentManager fragmentManager = getFragmentManager();
```

Comment:
Might as well pull out the FragmentManager to an instance variable since we'll be using it fairly often.